test/controllers/add.trust.controller.spec.ts

```diff
@@ -153,70 +153,71 @@ describe("Add Trust Controller Tests", () => {
 
       mockIsActiveFeature.mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_REDIS_REMOVAL
 
-      get(mockReq, mockRes, mockNext);
+      await get(mockReq, mockRes, mockNext);
 
       expect(mockNext).toBeCalledTimes(1);
       expect(mockNext).toBeCalledWith(error);
     });
   });
 
   describe('POST unit tests', () => {
-    test('select yes - Yes - for the current filing period to add trust', () => {
+    test('select yes - Yes - for the current filing period to add trust', async () => {
       (getApplicationData as jest.Mock).mockReturnValue(mockAppData);
       (generateTrustId as jest.Mock).mockReturnValue(trustId);
 
       mockReq.body = {
         addTrust: 'addTrustYes',
       };
 
-      post(mockReq, mockRes, mockNext);
+      await post(mockReq, mockRes, mockNext);
 
       expect(mockRes.redirect).toBeCalledTimes(1);
       expect(mockRes.redirect).toBeCalledWith(`${config.TRUST_DETAILS_URL}/${trustId}`);
     });
-    test('select Yes - for the pre-registration period to add trust', () => {
+    test('select Yes - for the pre-registration period to add trust', async () => {
       (getApplicationData as jest.Mock).mockReturnValue(mockAppData);
       (generateTrustId as jest.Mock).mockReturnValue(trustId);
 
       mockReq.body = {
         addTrust: 'preRegistration',
       };
 
-      post(mockReq, mockRes, mockNext);
+      await post(mockReq, mockRes, mockNext);
 
       expect(mockRes.redirect).toBeCalledTimes(1);
       expect(mockRes.redirect).toBeCalledWith(`${config.TRUST_DETAILS_URL}/${trustId}?relevant-period=true`);
     });
-    test('catches post request errors', () => {
+    test('catches post request errors', async () => {
       (getApplicationData as jest.Mock).mockReturnValue(mockAppData);
       (generateTrustId as jest.Mock).mockReturnValue(trustId);
 
-      post(mockReq, {} as Response, mockNext);
+      await post(mockReq, {} as Response, mockNext);
       expect(mockNext).toBeCalledTimes(1);
     });
 
-    test('select yes to add trust', () => {
+    test('select yes to add trust', async () => {
       mockReq.body = {
         addTrust: '0',
       };
 
-      post(mockReq, mockRes, mockNext);
+      await post(mockReq, mockRes, mockNext);
 
       expect(mockRes.redirect).toBeCalledTimes(1);
       expect(mockRes.redirect).toBeCalledWith(`${config.CHECK_YOUR_ANSWERS_URL}`);
     });
   });
 
   describe('POST unit tests with url params', () => {
-    test('select yes to add trust with url params', () => {
+    // ASM-350 - see failing reason
```

Comment:
What is the reason for failure? If the test is still applicable then it should pass. Not keen on committing skipped tests to the main branch.

Reply:
I forgot it, I have to see if the test is correct or if there is a problem

Reply:
I have checked and it is normal for this test to fail, it should never have passed.
The getUrlWithParamsToPath function is only called when updating. And in this case, we are simulating a registration.